src/components/Products/Table/index.js

```diff
@@ -0,0 +1,107 @@
+    const { itemsInCart, setCartItems, setCurrentProduct } = useCartContext()
+
+    const handleControls = (operation, productId) => {
+        const product = itemsInCart.find((element) => {
+            return element.id === productId
+        })
+        const productQuantity = product['quantity'];
+        const stock =  product['data'].stock
+        const quantityUpdated = getQuantity({ operation, stock, productQuantity})
+        product['operation'] = OPERATIONS.UPDATE
+        product['quantity']  =  quantityUpdated
+
+        setCurrentProduct(product)
+
+        setCartItems(prevCartItem => {
+            let cartItemUpdated = (operation === OPERATIONS.ADD) ? prevCartItem + GENERAL.SINGLE_ELEMENT : prevCartItem - GENERAL.SINGLE_ELEMENT
+            
+            if ( operation === OPERATIONS.REMOVE){
+                cartItemUpdated = prevCartItem - productQuantity
+            }
+
+            return cartItemUpdated
+        })
+    }
+
+
+
+    const getQuantity = ({ operation, stock, productQuantity}) => {
+
+        const quantityOperation = {
+            'add': function () {
+            return productQuantity < stock ? productQuantity + GENERAL.SINGLE_ELEMENT : stock;;
+            },
+            'rest': function () {
+            return productQuantity > GENERAL.PRODUCT_EMPTY ? productQuantity - GENERAL.SINGLE_ELEMENT : GENERAL.PRODUCT_EMPTY;
+            },
+            'remove': function () {
+            return  GENERAL.PRODUCT_EMPTY;
+            }
+        };
+
+        return quantityOperation[operation]();
+    }
+
+
+    return (
+
+            <StyledTable>
+                <thead>
+                    <tr>
+                        <th>Product</th>
+                        <th>Price</th>
+                        <th>Quantity</th>
+                        <th>Subtotal</th>
+                        <th className={checkout ? "hide": ""}>Remove product</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    { itemsInCart.map((item) => (
+
+                        item.quantity > GENERAL.PRODUCT_EMPTY &&
+                        <tr key={item.id}>
+                            <th>
+                                <Link  className="table__product"  to={`/product/${item.id}`}>
+                                    <img className="table__image" src={item.data.mainimage.url} alt={item.data.mainimage.alt} />
+                                    <span>{item.data.name}</span>                                
+                                </Link>
+                            </th>
+                            <th>{item.data.price.toFixed(GENERAL.DECIMAL_TO_SHOW)}</th>
+                            <th className="table__quantity">
+                                <Wrapper className={checkout ? "hide": ""}>
+                                    <QuantitySelector stock={item.data.stock} quantity={item.quantity} productId={item.id} handleControls={handleControls}/>
```

Comment:
cuando los props hacen que el uso de un componente se convierta en una linea muy larga, podemos separarlo como:

```suggestion
                                    <QuantitySelector
                                        prop={value}
                                        stock={stock}
                                        quantity={quantity}
                                        productId={id}
                                        handleControls={handleControls}
                                    />
```